src/Decompiler/Loading/Loader.cs

```diff
@@ -313,43 +385,46 @@ public Program LoadRawImage(string filename, byte[] image, Address? addrLoad, Lo
         /// </summary>
         /// <param name="fileName"></param>
         /// <returns>a TypeLibrary instance, or null if the format of the file wasn't recognized.</returns>
-        public TypeLibrary? LoadMetadata(string fileName, IPlatform platform, TypeLibrary typeLib)
+        public TypeLibrary? LoadMetadata(ImageLocation metadataLocation, IPlatform platform, TypeLibrary typeLib)
         {
-            var rawBytes = LoadImageBytes(fileName, 0);
-            var mdLoader = FindImageLoader<MetadataLoader>(fileName, rawBytes);
+            var rawBytes = LoadImageBytes(metadataLocation, 0);
+            var mdLoader = FindImageLoader<MetadataLoader>(metadataLocation, rawBytes);
             if (mdLoader is null)
                 return null;
             var result = mdLoader.Load(platform, typeLib);
             return result;
         }
 
-        public ScriptFile? LoadScript(string fileName)
+        public ScriptFile? LoadScript(ImageLocation scriptLocation)
         {
-            var rawBytes = LoadImageBytes(fileName, 0);
-            return FindImageLoader<ScriptFile>(fileName, rawBytes);
+            var rawBytes = LoadImageBytes(scriptLocation, 0);
+            return FindImageLoader<ScriptFile>(scriptLocation, rawBytes);
         }
 
         /// <summary>
         /// Loads the contents of a file with the specified filename into an array 
         /// of bytes, optionally at the offset <paramref>offset</paramref>. No interpretation
         /// of those bytes is done.
         /// </summary>
-        /// <param name="fileName">File to open.</param>
+        /// <param name="imageLocation">Location of the image whose bytes are to be loaded.</param>
         /// <param name="offset">The offset into the array into which the file will be loaded.</param>
         /// <returns>An array of bytes with the file contents at the specified offset.</returns>
-        public virtual byte[] LoadImageBytes(string fileName, int offset)
+        public virtual byte[] LoadImageBytes(ImageLocation imageLocation, int offset)
         {
+            // The initial fragment is always a file system location.
             var fsSvc = Services.RequireService<IFileSystemService>();
-            return fsSvc.ReadAllBytes(fileName);
+            return fsSvc.ReadAllBytes(imageLocation.FilesystemPath);
```

Comment:
If we always ignore archive entry fragments what the reason of change for `fileName` to `ImageLocation`. Maybe it would be better to leave it as `public virtual byte[] LoadImageBytes(string fileName, int offset)`?

Reply:
Hm, something got lost along the way here. The intent was to "drill into" the binary, expanding archives along the way. Perhaps this method no longer needs to exist. Loading a file either is recognized as an executable format, an archive format, or a blob. If it is an executable format, we're done. otherwise each archive format is responsible for extracting the bytes in whatever way is correct.

I'll review the codebase -- it's likely the `LoadImageBytes` method may disappear.

Reply:
`Loader` should now be clearer. There is a `Load` method which tries to first open a file as a `Project`, and if it isn't it opens it using `LoadBinaryImage`. If there are fragments, it attempts to 'drill' through one archive for each fragment until the end of the fragments are reached.

Reply:
Discussion of `LoadImageBytes` to follow.